### PR TITLE
fix(update): insert module meta relation if exist

### PR DESCRIPTION
**Pull request by lpinsivy**
## Description

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.04.x
- [x] 20.10.x
- [x] 21.04.x
- [x] 21.10.x (master)

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: www/install/php/Update-21.04.2.php [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    /**
+     * Retreive Meta Host Id
+     */
+    $statement = $pearDB->query(
+        "SELECT `host_id` FROM `host` WHERE `host_name` = '_Module_Meta'"
+    );
+    $result = $statement->fetch()
+    $moduleMetaId = $result['host_id'];
+
+    /*
```
Comment (kduret) [inline www/install/php/Update-21.04.2.php:47]:
```suggestion
    /*
```

File: www/install/php/Update-21.04.2.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+     * Add missing releation
+     */
+    if (is_int($moduleMetaId)) {
```
Comment (kduret) [inline www/install/php/Update-21.04.2.php:50]:
```suggestion

    /*
     * Add missing relation
     */
    if ($moduleMeta = $statement->fetch()) {
        $moduleMetaId = $moduleMeta['host_id'];
```

File: www/install/php/Update-21.04.2.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+ */
+try {
+    $pearDB->beginTransaction();
+    /**
+     * Retreive Meta Host Id
+     */
+    $statement = $pearDB->query(
+        "SELECT `host_id` FROM `host` WHERE `host_name` = '_Module_Meta'"
+    );
+    $result = $statement->fetch()
```
Comment (kduret) [inline www/install/php/Update-21.04.2.php:44]: missing `;`

File: www/install/php/Update-21.04.2.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+include_once __DIR__ . "/../../class/centreonLog.class.php";
+$centreonLog = new CentreonLog();
+
+//error specific content
+$versionOfTheUpgrade = 'UPGRADE - 21.04.2: ';
+
+$pearDB = new CentreonDB('centreon', 3, false);
+
+-- Add missing link between _Module_Meta and localhost poller
+
```
Comment (kduret) [inline www/install/php/Update-21.04.2.php:31]:
```suggestion
// Add missing link between _Module_Meta and localhost poller
```

File: www/install/php/Update-21.04.2.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+     */
+    if (is_int($moduleMetaId)) {
+        $errorMessage = "Unable to add relation between Module Meta and default poller.";
+        $statement = $pearDB->prepare(
+            "INSERT INTO ns_host_relation(`nagios_server_id`, `host_host_id`)
+            VALUES(
+                (SELECT id FROM nagios_server WHERE localhost = '1'),
+                (:moduleMetaId)
+            )
+            ON DUPLICATE KEY UPDATE nagios_server_id = (SELECT id FROM nagios_server WHERE localhost = '1')"
```
Comment (kduret) [inline www/install/php/Update-21.04.2.php:58]:
```suggestion
            ON DUPLICATE KEY UPDATE nagios_server_id = nagios_server_id"
```

**Review by kduret — CHANGES_REQUESTED** [review]

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Review by callapa — APPROVED** [review]

**Pull request merged**